vnext/Microsoft.ReactNative.Cxx/StructInfo.h

```diff
@@ -23,12 +34,25 @@
     fieldMap.emplace(fieldName, &TClass::field);                           \
   }
 
-#define INTERNAL_REACT_FIELD_1_ARGS(field) INTERNAL_REACT_FIELD_2_ARGS(field, L## #field)
-#define INTERNAL_REACT_FIELD_3RD_ARG(arg1, arg2, arg3, ...) arg3
-#define INTERNAL_REACT_FIELD_RECOMPOSER2(argsWithParentheses) INTERNAL_REACT_FIELD_3RD_ARG argsWithParentheses
+#define INTERNAL_REACT_FIELD_1_ARG(field) INTERNAL_REACT_FIELD_2_ARGS(field, L## #field)
 #define INTERNAL_REACT_FIELD(...) \
-  INTERNAL_REACT_FIELD_RECOMPOSER2((__VA_ARGS__, INTERNAL_REACT_FIELD_2_ARGS, INTERNAL_REACT_FIELD_1_ARGS, ))
-
+  INTERNAL_REACT_RECOMPOSER_3((__VA_ARGS__, INTERNAL_REACT_FIELD_2_ARGS, INTERNAL_REACT_FIELD_1_ARG, ))
+
+// REACT_STRUCT(structType)
+// Arguments:
+// - structType (required) - the struct name the macro is attached to.
+//
+// REACT_STRUCT annotates a C++ struct that then can be serialized and deserialized with IJSValueReader and
```

Comment:
>REACT_STRUCT [](start = 3, length = 12)

Nit: Don't know if this is universally applicable, but a key example to what each one of these macros expands might make all this "magic" less cryptic :-)